### ISSUE TITLE
Different variance in publication and implementation

### ISSUE THREAD

**Issue by glogowski-wojciech** [JianGoForIt/YellowFin_Pytorch#14]
In the publication the variance in "**Algorithm 3** Gradient variance" is defined as:
![image](https://user-images.githubusercontent.com/9470107/29736953-5e7c2e6e-89bb-11e7-90da-817b089d7c37.png)
However in the PyTorch implementation variance is defined as:
![image](https://user-images.githubusercontent.com/9470107/29736969-942c5a70-89bb-11e7-8e8f-17f3fef0a0b1.png)
Did you try YellowFin with variance from the publication? Were the results worse? Which definition of the variance did you use to produce results from the publication?

**Comment by JianGoForIt** [comment]
Hey @glogowski-wojciech,

Thanks for trying out YF. Actually, both in the experiment in the paper and in the repo, we use the same thing. The two formulations you give are the same as long as we do not use zero debias. In algorithm 3, you will see, as we defined, || \bar{g^2} ||_1 is the running average of squared gradient norm. || \bar{g}^2 ||_1 is the squared norm of the running average of gradient. Without zero-debits, you will see the corresponding elements in \bar{g^2} is larger than \bar{g}^2 and the elements in \bar{g^2} - \bar{g}^2 is always positive. That is why I say these two formulations are the same.

When we use zero-debasing, the two formulation might be different because, with zero debias, we can not guarantee \bar{g^2} is larger than \bar{g}^2. We don't expect this difference would trigger observable difference.
